vllm/model_executor/models/siglip_so400m.py

```diff
@@ -0,0 +1,543 @@
+
+import torch
+from PIL import Image
+from torch import nn
+from transformers import (BatchFeature, SiglipImageProcessor, SiglipProcessor,
+                          SiglipTextConfig, SiglipTokenizer,
+                          SiglipVisionConfig)
+
+from vllm.config import VllmConfig
+from vllm.model_executor.layers.linear import ColumnParallelLinear
+from vllm.model_executor.layers.pooler import DispatchPooler, Pooler
+from vllm.model_executor.layers.quantization import QuantizationConfig
+from vllm.model_executor.layers.vocab_parallel_embedding import (
+    VocabParallelEmbedding)
+from vllm.model_executor.model_loader.weight_utils import default_weight_loader
+from vllm.multimodal import MULTIMODAL_REGISTRY
+from vllm.multimodal.inputs import (MultiModalDataDict, MultiModalFieldConfig,
+                                    MultiModalKwargs)
+from vllm.multimodal.parse import MultiModalDataItems
+from vllm.multimodal.processing import (BaseMultiModalProcessor,
+                                        PromptIndexTargets, PromptInsertion,
+                                        PromptUpdate)
+from vllm.multimodal.profiling import BaseDummyInputsBuilder
+
+from .idefics2_vision_model import (Idefics2VisionAttention,
+                                    Idefics2VisionEmbeddings,
+                                    Idefics2VisionMLP)
+from .interfaces import SupportsMultiModal, default_pooling_type
+from .siglip import SiglipEncoderInfo
+from .utils import WeightsMapper
+
+if typing.TYPE_CHECKING:
+    from vllm.multimodal.processing import InputProcessingContext
+    from vllm.multimodal.utils import PlaceholderFeaturesInfo
+
+
+class SiglipSo400mProcessingInfo(SiglipEncoderInfo):
+
+    def __init__(self, ctx: "InputProcessingContext") -> None:
+        self.ctx = ctx
+        self.model_id = ctx.model_config.model
+        hf_config = ctx.get_hf_config()
+        self.hf_config = hf_config
+        self.vision_config = hf_config.vision_config
+
+    def get_hf_processor(self, **kwargs: Any) -> SiglipProcessor:
+        vision_config = self.hf_config
+        if hasattr(vision_config, "vision_config"):
+            vision_config = vision_config.vision_config
+        image_processor = SiglipImageProcessor(size={
+            "height":
+            vision_config.image_size,
+            "width":
+            vision_config.image_size
+        }, )
+        tokenizer_path = self.ctx.model_config.tokenizer
+        tokenizer = SiglipTokenizer.from_pretrained(tokenizer_path, **kwargs)
+        return SiglipProcessor(image_processor=image_processor,
+                               tokenizer=tokenizer)
+
+    def get_tokenizer(self) -> SiglipTokenizer:
+        return self.get_hf_processor().tokenizer
+
+    def get_allowed_mm_limits(self) -> Mapping[str, Optional[int]]:
+        return {"image": sys.maxsize}
+
+    def get_supported_mm_limits(self) -> Mapping[str, Optional[int]]:
+        return {"image": sys.maxsize}
```

Comment:
```suggestion
    def get_supported_mm_limits(self) -> Mapping[str, Optional[int]]:
        return {"image": None}
```

`None` means no limit already

Reply:
Hi @DarkLight1337 , Thanks for the feedback. I agree `None` is the better convention.

When I implemented it, run test commands, it uncovered a `TypeError` during engine initialization: `TypeError: '>' not supported between instances of 'NoneType' and 'int'`. This seems to originate from a `limit > 0` check in `vllm/multimodal/registry.py`.

My hypothesis is that this code path is specifically triggered for **pooling models** like this one. This would explain why other generative models can use `None` without issue.

I want to know if you have another suggestion.